### ISSUE TITLE
stringop-truncation warnings

### ISSUE THREAD

**Issue by PandorasFox** [Raymo111/i3lock-color#97]
```../../i3lock-color/unlock_indicator.c: In function ‘draw_image’:
../../i3lock-color/unlock_indicator.c:757:21: warning: ‘strncpy’ specified bound 40 equals destination size [-Wstringop-truncation]
                     strncpy(draw_data.status_text.str, noinput_text, sizeof(draw_data.status_text.str));
                     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../../i3lock-color/unlock_indicator.c:740:17: warning: ‘strncpy’ specified bound 40 equals destination size [-Wstringop-truncation]
                 strncpy(draw_data.status_text.str, wrong_text, sizeof(draw_data.status_text.str));
                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../../i3lock-color/unlock_indicator.c:724:17: warning: ‘strncpy’ specified bound 40 equals destination size [-Wstringop-truncation]
                 strncpy(draw_data.status_text.str, verif_text, sizeof(draw_data.status_text.str));
                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../../i3lock-color/unlock_indicator.c:784:9: warning: ‘strncpy’ specified bound 40 equals destination size [-Wstringop-truncation]
         strncpy(draw_data.mod_text.str, modifier_string, sizeof(draw_data.mod_text.str));
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../../i3lock-color/unlock_indicator.c:794:9: warning: ‘strncpy’ specified bound 40 equals destination size [-Wstringop-truncation]
         strncpy(draw_data.keylayout_text.str, layout_text, sizeof(draw_data.keylayout_text.str));
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Looks like the draw_data buffers need to have their size increased by one or so, somewhere. Thanks dank new gcc8 string protections for the warnings; I'll have to poke at it later.

**Comment by PandorasFox** [comment]
```┌arcana@lapfox-(~/i3lock-color)  [130] 
└> cat ./asan_suppressions.txt 
# fontconfig....
leak:libfontconfig
┌arcana@lapfox-(~/i3lock-color)   
└> ls -al /tmp
┌arcana@lapfox-(~/i3lock-color)  [130] 
└> LSAN_OPTIONS=suppressions=/home/arcana/i3lock-color/asan_suppressions.txt ./lock.sh 
-----------------------------------------------------
Suppressions used:
  count      bytes template
     29       2849 libfontconfig
-----------------------------------------------------

┌arcana@lapfox-(~/i3lock-color)   
└> -----------------------------------------------------
-----------------------------------------------------
Suppressions used:
Suppressions used:
  count      bytes template
  count      bytes template
     35       3473 libfontconfig
     29       2849 libfontconfig
-----------------------------------------------------

-----------------------------------------------------
```

Gonna just close this. I can suppress libfontconfig in ASAN and it's not something easily fixable (or something that impacts, well, anything).

**Comment by PandorasFox** [comment]
fixed in ce7c826acbeac04de675e9e14e2a62418c70c2c7
